src/xray/askui-xray-step-reporter.ts

```diff
@@ -0,0 +1,155 @@
+import { Reporter, Step, StepStatus, ReporterConfig } from 'askui';
+import { convertPngDataUrlToBase64 } from '../utils/image-reporting-utils';
+import path from 'path';
+import fs from 'fs';
+
+/*
+ * Possible XRay status: PASS, FAIL, TODO
+ */
+enum Status {
+  PASS = 'PASS',
+  FAIL = 'FAIL',
+  TODO = 'TODO',
+}
+
+type XRayTestObject = {
+  testKey: string;
+  status?: Status;
+  start?: string;
+  finish?: string;
+  steps?: XRayStep[];
+}
+
+type XRayEvidence = {
+  data: string;
+  filename: string;
+  contentType: string;
+}
+
+type XRayStep = {
+  status?: Status;
+  actualResult?: string;
+  evidences: XRayEvidence[];
+}
+
+function mapAskuiToXrayStepStatus(status: StepStatus): Status {
+  switch (status) {
+    case 'passed':
+      return Status.PASS;
+    case 'failed':
+      return Status.FAIL;
+    case 'erroneous':
+      return Status.FAIL;
+    default:
+      return Status.TODO;
```

Comment:
Is that a good default or is there a proper case we could use instead which makes this less likely to become an undetected error if we add another status in the future?

Reply:
According to the XRAY docs they do not have predefined status. So I went with this one. We could of course use something more useful. What would be your suggestion?

Reply:
I would just not make it a default but rather throw an error here so we notice that a status we don't know is used or use a special `UNDEFINED` status.